### PR TITLE
Added online debug log creator

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following (please check boxes, eg [X] - no spaces) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 10

---

Calls `sudo pihole -d -a`

screenshot at 2017-02-10 12-22-28

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

**Comment by dschaper** [comment]
`sudo pihole -d -a` is going to be the trigger then?

**Comment by DL6ER** [comment]
Yes. it can still be changed if you'd like to have something different

**Comment by dschaper** [comment]
LGTM.

**Comment by DL6ER** [comment]
Codacy is complaining about a bug that is not there in the most recent version if the code. I told it to re-analyze, but that seems to take forever.

**Comment by PromoFaux** [comment]
Can we not make it an optional upload still?

I'm thinking Modal dialog box with a "Would you like to upload the debug log once it has completed?" before the script runs.

OR alternatively a button that will just run the `nc` command to upload `/var/log/pihole_debug.log` at the end?

**Comment by DL6ER** [comment]
> I'm thinking Modal dialog box with a "Would you like to upload the debug log once it has completed?" before the script runs.

Yeah, that is possible, depending on that we can than decide whether to call `pihole -d` or `pihole -d -a`

**Comment by dschaper** [comment]
`pihole -d` will still block though, it's going to wait for user interaction. There's probably going to be more flags needed to be passed, one for auto-no upload and one for auto-with upload. Unless there's another way around the issue.

**Comment by PromoFaux** [comment]
Maybe just lower the blocking time at the end of the script, take the interaction out of it and make it mandatory...

**Comment by dschaper** [comment]
It has the timeout now, so it will stop the log capture, but then it still requires a Y/N for log upload. If the solution is that `pihole -d` auto-uploads, then that's a problem.

**Comment by DL6ER** [comment]
> it still requires a Y/N for log upload

The pipe that is opened by our PHP script is read-only, hence `read` will fail (as if saying "No"). In fact, there is no problem:
- `pihole -d` won't try to upload,
- `pihole -d -a` will try to upload.

**Comment by dschaper** [comment]
Yeah, I see now that it's no-TTY, so `read` fails, the script sees that as a non=`y` answer and doesn't attemtp to upload.

File: scripts/pi-hole/js/debug.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    ta.empty();
+    ta.show();
+
+    source.addEventListener("message", function(e) {
+        ta.append(e.data);
+    }, false);
+
+    // Will be called when script has finished
+    source.addEventListener("error", function(e) {
+        source.close();
```
Comment (Mcat12) [inline scripts/pi-hole/js/debug.js]: Should re-enable the button and checkbox when finished.

Comment (dschaper) [inline scripts/pi-hole/js/debug.js]: I'd like to keep that as it is. Force the users to at least reload the page before sending another debug.

**Comment by dschaper** [comment]
Approved

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/pi-hole-2-13-core-2-5-web/2004/1

**Pull request merged**